jobbergate-api/jobbergate_api/pagination.py

```diff
@@ -2,18 +2,89 @@
 Pagination feature for all endpoints.
 """
 
+from typing import Generic, List, Optional, TypeVar
 
-class Pagination:
+from pydantic import BaseModel, Field
+from pydantic.generics import GenericModel
+from sqlalchemy import func, select
+
+from jobbergate_api.storage import database
+
+
+class Pagination(BaseModel):
     """
     Basic pagination class.
     """
 
-    def __init__(self, limit: int = 10, skip: int = 0):
+    page: Optional[int] = Field(
```

Comment:
It needs to be optional so that if the user doesn't pass any pagination parameters, we know that they are requesting all the data. Having a value of 0 here means that you are getting just the first page. The CLI, for instance, doesn't use pagination (yet), so we need to be able to skip pagination if it is not requested.